src/core/mac/mac_frame.cpp

```diff
@@ -172,122 +172,11 @@ uint16_t Frame::GetFrameControlField(void) const
 
 otError Frame::ValidatePsdu(void) const
 {
-    otError error = OT_ERROR_PARSE;
-    uint8_t offset = kFcfSize + kDsnSize;
-    uint8_t footerLength = kFcsSize;
-    uint16_t fcf;
-
-    VerifyOrExit((offset + footerLength) <= GetPsduLength());
-
-    fcf = GetFrameControlField();
-
-    // Destination PAN + Address
-    switch (fcf & kFcfDstAddrMask)
-    {
-    case kFcfDstAddrNone:
-        break;
-
-    case kFcfDstAddrShort:
-        offset += sizeof(PanId) + sizeof(ShortAddress);
-        break;
-
-    case kFcfDstAddrExt:
-        offset += sizeof(PanId) + sizeof(ExtAddress);
-        break;
-
-    default:
-        goto exit;
-    }
-
-    // Source PAN + Address
-    switch (fcf & kFcfSrcAddrMask)
-    {
-    case kFcfSrcAddrNone:
-        break;
-
-    case kFcfSrcAddrShort:
-        if ((fcf & kFcfPanidCompression) == 0)
-        {
-            offset += sizeof(PanId);
-        }
-
-        offset += sizeof(ShortAddress);
-        break;
-
-    case kFcfSrcAddrExt:
-        if ((fcf & kFcfPanidCompression) == 0)
-        {
-            offset += sizeof(PanId);
-        }
-
-        offset += sizeof(ExtAddress);
-        break;
-
-    default:
-        goto exit;
-    }
-
-    VerifyOrExit((offset + footerLength) <= GetPsduLength());
```

Comment:
Should include this length check as well.

Reply:
Added this check to` FindPayloadIndex()`, thanks.